lib/api/2.0/nodes.js

```diff
@@ -61,24 +62,45 @@ var nodesGetPollersById = controller(function(req) {
 });
 
 var nodesGetWorkflowById = controller(function(req) {
-    return nodes.getNodeWorkflowById(req.swagger.params.identifier.value);
+    return workflowApiService.getWorkflowsByNodeId(req.swagger.params.identifier.value,
+                                                   req.query)
+        .then(function(workflows){
+            if(req.swagger.query && _(req.swagger.query).has('active')) {
+                return _(workflows)
+                    .filter(function(workflow) {
+                        return workflow.active() === req.swagger.query.active;
```

Comment:
Couldn't we just update the getWorkflowsByNodeId to take an `active` option that just returns zero or one documents, rather than getting the entire workflow history (which could be quite large) and iterating through it?